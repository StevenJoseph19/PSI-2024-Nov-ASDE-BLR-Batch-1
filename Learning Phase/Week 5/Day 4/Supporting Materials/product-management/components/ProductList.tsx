import React from 'react';

type Product = { id: number; name: string; price: string };

const ProductList: React.FC<{ products: Product[] }> = ({ products }) => {
  return (
    <ul>
      {products.map((product) => (
        <li key={product.id}>
          {product.name} - ${product.price}
        </li>
      ))}
    </ul>
  );
};

export default ProductList;
