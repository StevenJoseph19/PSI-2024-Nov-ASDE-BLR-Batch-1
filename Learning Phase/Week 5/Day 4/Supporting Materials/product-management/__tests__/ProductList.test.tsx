import { render, screen } from '@testing-library/react';
import ProductList from '../components/ProductList';
import '@testing-library/jest-dom';

describe('ProductList Component', () => {
  it('renders a list of products', () => {
    const products = [
      { id: 1, name: 'Product 1', price: '10.00' },
      { id: 2, name: 'Product 2', price: '20.00' },
    ];

    render(<ProductList products={products} />);

    // Check if the products are rendered
    expect(screen.getByText(/product 1/i)).toBeInTheDocument();
    expect(screen.getByText(/10.00/i)).toBeInTheDocument();
    expect(screen.getByText(/product 2/i)).toBeInTheDocument();
    expect(screen.getByText(/20.00/i)).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const products = [
      { id: 1, name: 'Product 1', price: '10.00' },
      { id: 2, name: 'Product 2', price: '20.00' },
    ];
    const { asFragment } = render(<ProductList products={products} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
