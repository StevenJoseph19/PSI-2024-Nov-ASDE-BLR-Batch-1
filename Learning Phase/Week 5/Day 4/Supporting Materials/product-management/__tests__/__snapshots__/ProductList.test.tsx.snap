// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList Component matches snapshot 1`] = `
<DocumentFragment>
  <ul>
    <li>
      Product 1 - $10.00
    </li>
    <li>
      Product 2 - $20.00
    </li>
  </ul>
</DocumentFragment>
`;
